modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/repository_entry_browser/js/ItemSelectorRepositoryEntryBrowser.es.js

```diff
@@ -105,7 +97,28 @@ class ItemSelectorRepositoryEntryBrowser extends PortletBase {
 
 		this._itemSelectorUploader.destroy();
 
-		this._eventHandler.removeAllListeners();
+		if (this._itemPreviewClickHandle) {
+			this._itemPreviewClickHandle.dispose();
+		}
+
+		if (this._inputFileNode) {
+			this._inputFileNode.removeEventListener(
+				'change',
+				this._handleInputFileNodeChange
+			);
+		}
+
+		if (this.rootNode) {
+			this.rootNode.removeEventListener(STR_DRAG_OVER, (event) =>
+				this._ddEventHandler(event)
+			);
+			this.rootNode.removeEventListener(STR_DRAG_LEAVE, (event) =>
+				this._ddEventHandler(event)
+			);
+			this.rootNode.removeEventListener(STR_DROP, (event) =>
+				this._ddEventHandler(event)
+			);
```

Comment:
`removeEventListener` requires that you pass in _the exact same_ listener that you subscribed with, but all of these functions are distinct, so these calls won't work at all. Try this in the JS console:

```js
((event) => this._ddEventHandler(event)) === ((event) => this._ddEventHandler(event)) // false
```

However if you instead do this:

```js
rootNode.addEventListener(STR_DROP, this._ddEventHandler);
rootNode.removeEventListener(STR_DROP, this._ddEventHandler);
```

then the addition/removal would work, because you'd be passing in exactly the same function.

You can't do that though, because `_ddEventHandler()` relies on `this` being correctly set, so you'll need to find some other way to make sure the function you subscribe and remove is stable.

Reply:
Alrighty, that's the one module I couldn't manage to test the `detach()` of, so it's understandable that it's buggy. I'll take a look at that tomorrow, thanks!